barman/cloud_providers/google_cloud_storage.py

```diff
@@ -333,3 +337,282 @@ def _delete_objects_batch(self, paths):
+                        disk_name,
+                    ),
+                },
+            }
+        )
+
+        _logger.info("Waiting for snapshot '%s' completion", snapshot_name)
+        resp.result()
+
+        if resp.done():
+            _logger.info("Snapshot '%s' completed", snapshot_name)
+            return snapshot_name
+        else:
+            raise CloudProviderError(
+                "Snapshot '%s' failed with error code %s: %s"
+                % (snapshot_name, resp.error_code, resp.error_message)
+            )
+
+    def take_snapshot_backup(self, backup_info, instance_name, zone, disks):
+        """
+        Take a snapshot backup for the named instance.
+
+        Creates a snapshot for each named disk and saves the required metadata
+        to snapshot_info.
+
+        The following additional field is saved to snapshots_info:
+
+            * gcp_project: The name of the project to which all snapshot-related
+                           resources belong
+
+        The following additional field is saved in the metadata dict for each snapshot:
+
+            * physical_block_size_bytes: The block size of the source disk in bytes.
+            * device_name: The name of the device associated with the source disk, as
+                           understood by the GCP API.
+            * size_gb: The size of the source disk in GB.
+
+        :param barman.infofile.LocalBackupInfo backup_info: Backup information.
+        :param str instance_name: The name of the VM instance to which the disks
+            to be backed up are attached.
+        :param str zone: The zone in which the snapshot disks and instance reside.
+        :param list[str] disks: A list containing the names of the source disks.
+        """
+        instance_metadata = self._get_instance_metadata(instance_name, zone)
+        snapshots = []
+        for disk_name in disks:
+            disk_metadata = self._get_disk_metadata(disk_name, zone)
+            metadata = {
+                "block_size": disk_metadata.physical_block_size_bytes,
+                "size": disk_metadata.size_gb,
+            }
+            # Check disk is attached and find device name
+            attached_disks = [
+                d
+                for d in instance_metadata.disks
+                if d.source == disk_metadata.self_link
+            ]
+            if len(attached_disks) == 0:
+                raise SnapshotBackupException(
+                    "Disk %s not attached to instance %s" % (disk_name, instance_name)
+                )
+            # We should always have exactly one attached disk matching the name
+            assert len(attached_disks) == 1
+
+            metadata["device_name"] = attached_disks[0].device_name
+            metadata["device_path"] = self.DEVICE_PREFIX + attached_disks[0].device_name
+            metadata["name"] = self.take_snapshot(backup_info, zone, disk_name)
+            snapshots.append(metadata)
```

Comment:
@didiermichel I'm adding this comment so we remember to update the snapshots_info format as discussed on slack

Reply:
Resolving this as the snapshots_info metadata changes are done.